backend/src/models/db-questions.js

```diff
@@ -0,0 +1,82 @@
+const sqlite3 = require('sqlite3')
+const db = new sqlite3.Database('./db/questions.db')
+
+function get_all_questions(data) {
+  // Lookup all questions
+  return new Promise((resolve, reject) => { 
+    db.all('SELECT rowid, uid, title, body FROM questions', (err, rows) => {
+    
+      // Check for a qeury error
+      if (err) {
+        reject(data)
+      }
+
+      // Transform the requested data into the appropriate format
+      data = rows.map(({rowid, uid, title, body}) => ({
+        id: rowid,
+        meta: {uid},
+        title,
+        body
+      }))
+      resolve(data)
+    })
+  })
+}
+
+function get_question(data, question_id) {
+  // Lookup a specific question
+  return new Promise((resolve, reject) => {
+    db.get(
+      'SELECT rowid, uid, title, body FROM questions WHERE rowid=$qid',
+      { $qid: question_id },
+      (err, row) => {
+        
+        // Check for a qeury error
+        if (err) {
+          reject(data)
+        }
+
+        // Transform the requested data into the appropriate format
```

Comment:
This is sort of accidental in how it works out. Better probably to just return the question as is.

```js
resolve({
          id: row.rowid,
          meta: {uid: row.uid},
          title: row.title,
          body: row.body
        })
```

Reply:
Not quite sure I understand what you mean here? I've adjusted it to the code snippet you provided though.